Stats - Add the published module

daniloercoli opened the pull request:
#2494 Add the "Published" module to Stats. 

The module is designed to be in par with the web version.

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java:
```diff
@@ -0,0 +1,173 @@
+        return !isDataEmpty() && ((PublishedPostsAndPagesModel) mDatamodels[0]).hasPublishedPostsAndPages();
+    }
+
+    private List<PostModel> getTopPostsAndPages() {
```
Should this be renamed to something like mPublishedPostsAndPages?

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java:
```diff
@@ -0,0 +1,173 @@
+
+        private List<PostModel> list;
+        private final LayoutInflater inflater;
+        private final int localTableBlogID;
```
This field is unused. Will it be needed in the future or is it worth taking out?

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java:
```diff
@@ -0,0 +1,68 @@
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.JSONUtils;
```
This import and Iterator can be removed.

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java:
```diff
@@ -0,0 +1,68 @@
+
+public class PublishedPostsAndPagesModel implements Serializable {
+    private final String mBlogID;
+    private List<PostModel> mTopPostsAndPages;
```
Should this be renamed to something like mPublishedPostsAndPages?

tonyr59h commented:
Tested different blogs, different time spans, on Nexus 5 and S3. Looks good and works as expected.

I notice that there's a bit more to the web implementation of the stats views. It's collapsable and has an info icon:
![](https://cloud.githubusercontent.com/assets/518232/6966720/6a5949c8-d95b-11e4-8a06-baf55b42f4ee.png) 

Are there plans to bring these views to the mobile platforms?

daniloercoli commented:
> Are there plans to bring these views to the mobile platforms?

At the beginning of Calypso-Stats we decided to not implement collapsable modules on mobile. And we don't have that feature scheduled in the short term.  I will bring this up during the next Stats squad chat.

daniloercoli commented:
@tonyr59h - Renamed unused variables and removed unused fields and imports. That weird though, I remember I had already renamed those fields. I probably missed to commit those changes.

tonyr59h commented:
:shipit:
